eacl/target.go

```diff
@@ -1,200 +1,174 @@
 package eacl
 
 import (
-	"bytes"
-	"crypto/ecdsa"
+	"fmt"
 
-	"github.com/nspcc-dev/neo-go/pkg/crypto/keys"
+	"github.com/nspcc-dev/neo-go/pkg/util/slice"
 	v2acl "github.com/nspcc-dev/neofs-api-go/v2/acl"
+	neofscrypto "github.com/nspcc-dev/neofs-sdk-go/crypto"
 )
 
-// Target is a group of request senders to match ContainerEACL. Defined by role enum
-// and set of public keys.
-//
-// Target is compatible with v2 acl.EACLRecord.Target message.
+// Target describes the parties that are subject to a specific access rule.
 type Target struct {
-	role Role
-	keys [][]byte
-}
-
-func ecdsaKeysToPtrs(keys []ecdsa.PublicKey) []*ecdsa.PublicKey {
-	keysPtr := make([]*ecdsa.PublicKey, len(keys))
-
-	for i := range keys {
-		keysPtr[i] = &keys[i]
-	}
-
-	return keysPtr
+	roles []Role
+	keys  [][]byte
 }
 
-// CopyTo writes deep copy of the [Target] to dst.
-func (t Target) CopyTo(dst *Target) {
-	dst.role = t.role
-
-	dst.keys = make([][]byte, len(t.keys))
-	for i, k := range t.keys {
-		dst.keys[i] = make([]byte, len(k))
-
-		copy(dst.keys[i], t.keys[i])
-	}
-}
-
-// BinaryKeys returns list of public keys to identify
-// target subject in a binary format.
+// NewTarget returns Target that matches parties with at least one of the
+// specified roles or public keys.
 //
-// Each element of the resulting slice is a serialized compressed public key. See [elliptic.MarshalCompressed].
-// Use [neofsecdsa.PublicKey.Decode] to decode it into a type-specific structure.
+// At least one argument MUST be non-empty. Both arguments MUST NOT be mutated
+// within resulting Target lifetime. All roles MUST be supported values from the
+// corresponding enum except [RoleSystem] which is no longer supported. All keys
+// MUST be non-nil.
 //
-// The value returned shares memory with the structure itself, so changing it can lead to data corruption.
-// Make a copy if you need to change it.
-func (t *Target) BinaryKeys() [][]byte {
-	return t.keys
-}
+// See also other helper constructors.
+func NewTarget(roles []Role, publicKeys []neofscrypto.PublicKey) Target {
```

Comment:
again, why some version of a generic struct is more default than others? why keys are more default `New` than roles?